core/coreapi/unixfs.go

```diff
@@ -126,6 +131,23 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.File, opts ...options
 		fileAdder.SetMfsRoot(mr)
 	}
 
+	// Get storageMax when node is online.
+	if fileAdder.Pin && !n.EnableGC && n.OnlineMode() {
+		storageUsage, err := n.Repo.GetStorageUsage()
```

Comment:
Is `GetStorageUsage` a faithful metric of repo size? Is it compatible with `StorageMax`? (this is a naive question but I'm not familiar with the repo API)

Reply:
In my testing , maybe `GetStorageUsage ` is a faithful metric of repo size.